tensorflow/core/util/mkl_util.h

```diff
@@ -1159,29 +1529,44 @@ class MklDnnData {
       user_memory_ = new memory(pd);
     }
   }
+#endif
 
+#ifndef ENABLE_MKLDNN_V1
+  /// Memory primitive descriptor is deprecated in MKL-DNN v1.x
   /// A version of SetUsrMem with primitive descriptor and tensor
   inline void SetUsrMem(const memory::primitive_desc& pd,
                         const Tensor* tensor) {
     CHECK_NOTNULL(tensor);
     SetUsrMem(pd, GetTensorBuffer(tensor));
   }
+#endif
 
   /// Get function for user memory primitive.
   inline const memory* GetUsrMem() const { return user_memory_; }
 
+#ifndef ENABLE_MKLDNN_V1
+  /// Memory primitive descriptor is deprecated in MKL-DNN v1.0.
   /// Get function for primitive descriptor of user memory primitive.
   inline const memory::primitive_desc GetUsrMemPrimDesc() const {
     CHECK_NOTNULL(user_memory_);
     return user_memory_->get_primitive_desc();
   }
+#endif
 
+#ifdef ENABLE_MKLDNN_V1
+  /// Get function for descriptor of user memory.
+  inline memory::desc GetUsrMemDesc() const {
+    CHECK_NOTNULL(user_memory_);
```

Comment:
```suggestion
    DCHECK(user_memory_);
```
Same with other places.

Reply:
Done